src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,310 @@
+                                                  nullptr,
+                                                  nullptr);
+
+    NetSecurityNative_CopyBuffer(&gssBuffer, outBuffer);
+    return majorStatus;
+}
+
+extern "C" uint32_t NetSecurityNative_ReleaseCred(uint32_t* minorStatus, GssCredId** credHandle)
+{
+    assert(minorStatus != nullptr);
+    assert(credHandle != nullptr);
+
+    return gss_release_cred(minorStatus, credHandle);
+}
+
+extern "C" void NetSecurityNative_ReleaseGssBuffer(void* buffer, uint64_t length)
+{
+    assert(buffer != nullptr);
+
+    uint32_t minorStatus;
+    GssBuffer gssBuffer{.length = length, .value = buffer};
+    gss_release_buffer(&minorStatus, &gssBuffer);
+}
+
+extern "C" uint32_t NetSecurityNative_ReleaseName(uint32_t* minorStatus, GssName** inputName)
+{
+    assert(minorStatus != nullptr);
+    assert(inputName != nullptr);
+
+    return gss_release_name(minorStatus, inputName);
+}
+
+extern "C" uint32_t NetSecurityNative_Wrap(uint32_t* minorStatus,
+                                           GssCtxId* contextHandle,
+                                           int32_t isEncrypt,
+                                           uint8_t* inputBytes,
+                                           int32_t offset,
+                                           int32_t count,
+                                           struct PAL_GssBuffer* outBuffer)
+{
+    assert(minorStatus != nullptr);
+    assert(contextHandle != nullptr);
+    assert(isEncrypt == 1 || isEncrypt == 0);
+    assert(inputBytes != nullptr);
+    assert(offset >= 0);
+    assert(count >= 0);
+    assert(outBuffer != nullptr);
+    // count refers to the length of the input message. That is, number of bytes of inputBytes
+    // starting at offset
+    // that need to be wrapped.
+
+    int confState;
+    GssBuffer inputMessageBuffer{.length = UnsignedCast(count), .value = inputBytes + offset};
+    GssBuffer gssBuffer;
+    uint32_t majorStatus =
+        gss_wrap(minorStatus, contextHandle, isEncrypt, GSS_C_QOP_DEFAULT, &inputMessageBuffer, &confState, &gssBuffer);
+
+    NetSecurityNative_CopyBuffer(&gssBuffer, outBuffer);
+    return majorStatus;
+}
+
+extern "C" uint32_t NetSecurityNative_Unwrap(uint32_t* minorStatus,
+                                             GssCtxId* contextHandle,
+                                             uint8_t* inputBytes,
+                                             int32_t offset,
+                                             int32_t count,
+                                             struct PAL_GssBuffer* outBuffer)
+{
+    assert(minorStatus != nullptr);
+    assert(contextHandle != nullptr);
+    assert(inputBytes != nullptr);
+    assert(offset >= 0);
+    assert(count >= 0);
+    assert(outBuffer != nullptr);
+
+    // count refers to the length of the input message. That is, the number of bytes of inputBytes
+    // starting at offset  that need to be wrapped.
```

Comment:
Nit: there is an extra unnecessary space between "starting at offset" and "that need to be wrapped."